WalletWasabi.Gui/Controls/WalletExplorer/CoinListViewModel.cs

```diff
@@ -424,6 +424,7 @@ public void OnOpen()
 									if (toRemove != default)
 									{
 										RootList.Remove(toRemove);
+										toRemove.UnsubscribeEvents();
```

Comment:
The Unsubscription is happening elsewhere: `.OnItemRemoved(x => x.UnsubscribeEvents())`

Reply:
However if I use OnItemAdded instead of the method that you're using then interesting things happen, so maybe the OnItemRemoved should be removed and your method left.

Reply:
As far as I can remember strange things start happening if you use OnItemRemoved. In some cases there the items in the list messed up. Code should stay like this.